### PR TITLE
SDK-implementation-for-volume-attachments

### PR DESCRIPTION
**Pull request by adityas1netapp**
Checklist
---------

JIRA Ticket - [SPOTAUT-16756](https://spotinst.atlassian.net/browse/SPOTAUT-16756)
Checklist (Please mark Yes/No/NA for each of the questions below) -

- I have provided clear and detailed explanation of changes made in the proposed PR : No
- I have compiled the code successfully my laptop : No
- I have written examples to demonstrates the usage of the newly implemented APIs : No
- I have incremented the release version : No
- I have run the Intellij Code Analysis on newly added code and fixed all (valid) reported issues : No

OR

- I have taken approval from scrum master or project manager for skipping any of the items above : No

[SPOTAUT-16756]: https://spotinst.atlassian.net/browse/SPOTAUT-16756?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/model/ApiGroupCompute.java [hunk 1]
```diff
@@ -29,6 +30,7 @@ class ApiGroupCompute implements IPartialUpdateEntity {
     private List<String>        preferredAvailabilityZones;
     private List<String> privateIps;
     private List<String> subnetIds;
+    private ElastigroupVolumeAttachments volumeAttachments;
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ApiGroupCompute.java:33]: this should be API class not BL

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 2]
```diff
@@ -480,6 +480,10 @@ private static ApiGroupCompute toDal(ElastigroupComputeConfiguration compute) {
                 optCompute.setSubnetIds(compute.getSubnetIds());
             }
 
+            if(compute.isVolumeAttachmentsSet()){
+                optCompute.setVolumeAttachments(compute.getVolumeAttachments());
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:484]: toDal reference is missing

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 3]
```diff
@@ -1171,6 +1175,39 @@ private static ApiDynamicVolumeSize toDal(DynamicVolumeSize dynamicVolumeSize) {
...
+        ApiElastigroupVolumeAttachments retVal = null;
+
+        if (volumeAttachments != null) {
+            retVal = new ApiElastigroupVolumeAttachments();
+
+            if(volumeAttachments.isVolumesSet()){
+                List<ApiElastigroupVolume> apiVolumes = volumeAttachments.getVolumes().stream().map(ElastigroupConverter::toDal).collect(Collectors.toList());
+                retVal.setVolumes(apiVolumes);
+
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:1205]: remove extra line spaces

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 4]
```diff
@@ -2650,6 +2687,10 @@ private static ElastigroupComputeConfiguration toBl(ApiGroupCompute compute) {
                 blComputeBuilder.setSubnetIds(compute.getSubnetIds());
             }
 
+            if(compute.isVolumeAttachmentsSet()){
+                blComputeBuilder.setVolumeAttachments(compute.getVolumeAttachments());
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:2691]: use toBl reference

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 5]
```diff
@@ -2677,6 +2718,41 @@ private static ElastigroupEbsVolumePool toBl(ApiVolumePool volumePool) {
...
+        ElastigroupVolumeAttachments retVal = null;
+
+        if (volumeAttachments != null) {
+            ElastigroupVolumeAttachments.Builder retValBuilder = ElastigroupVolumeAttachments.Builder.get();
+            if (volumeAttachments.isVolumesSet()) {
+                List<ApiElastigroupVolume> apiVolumes = volumeAttachments.getVolumes();
+                List<ElastigroupVolume> volumes = apiVolumes.stream().map(ElastigroupConverter::toBl)
+                        .collect(Collectors.toList());
+                retValBuilder.setVolumes(volumes);
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:2748]: remove extra line spaces

File: src/main/java/com/spotinst/sdkjava/model/api/elastigroup/aws/ApiElastigroupVolume.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+    //region Members
+    @JsonIgnore
+    private Set<String> isSet;
+    private String deviceName;
+    private String volumeId;
+
+
+    //endregion
+
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/api/elastigroup/aws/ApiElastigroupVolume.java:30]: remove extra line spaces wherever necessary

File: src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/aws/ElastigroupVolume.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Created by Aditya on 11/10/23.
+ */
+
+public class ElastigroupVolume {
```
Comment (chandra1-n) [inline src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/aws/ElastigroupVolume.java:12]: change the classname to ElastigroupVolumes

**Review by chandra1-n — CHANGES_REQUESTED** [review]
please fix the comments

File: src/main/java/com/spotinst/sdkjava/model/ApiGroupCompute.java (hunk 1 above)
Comment (adityas1netapp) [reply src/main/java/com/spotinst/sdkjava/model/ApiGroupCompute.java:33]: changed

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java (hunk 3 above)
Comment (adityas1netapp) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:1205]: removed

File: src/main/java/com/spotinst/sdkjava/model/api/elastigroup/aws/ApiElastigroupVolume.java (hunk 6 above)
Comment (adityas1netapp) [reply src/main/java/com/spotinst/sdkjava/model/api/elastigroup/aws/ApiElastigroupVolume.java:30]: removed

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java (hunk 2 above)
Comment (adityas1netapp) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:484]: added

File: src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/aws/ElastigroupVolume.java (hunk 7 above)
Comment (adityas1netapp) [reply src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/aws/ElastigroupVolume.java:12]: changed

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java (hunk 2 above)
Comment (adityas1netapp) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java]: changed

**Review by chandra1-n — APPROVED** [review]

**Pull request merged**